### PR TITLE
Switch to Github Actions

### PR DESCRIPTION
**Pull request by rvolosatovs**
#### Summary

Travis CI has a hard limit of 50 minutes for builds. Our builds constantly time out and not only CI fails, but also the build does not get to the Docker registry.
Tried Docker autobuilds first - those do not time out(I believe the hard limit is 4 hours, but are twice as slow - builds take 2 hours to finish on average)
Github Actions are free, powerful and on par with Travis performance-wise(around 45-50 minutes), hence the switch. I understand this is still in beta, but we are not doing anything crazy here, so any kind of breaking changes from their side should not be too hard to fix. In (unlikely) case we find out that Github Actions do not fulfill our requirements we can always switch to the (slow) Docker autobuilds.(I have it working in a branch, just a matter of opening a PR)

#### Changes

- Switch to Github Actions
- Remove redundant `Makefile`
- Alter tagging scheme - we don't do `nightly` builds anymore.
   - `development` now corresponds to latest `master` build
   - `latest` now corresponds to latest non-ttn tag build
   - `latest-ttn` now corresponds to latest ttn tag build
   - rest is unchanged

#### Release Notes

- `nightly` builds are not going to be updated anymore
- `development` now corresponds to latest `master` build
- `latest` now corresponds to latest non-ttn tag build
- `latest-ttn` now corresponds to latest ttn tag build

### CODE REVIEW

**Comment by rvolosatovs** [comment]
You can see the relevant CI build [here](https://github.com/TheThingsIndustries/docker-protobuf/runs/279653825)

**Comment by rvolosatovs** [comment]
FTR, I received an email today stating that Github Actions will be out of beta in 2 weeks.

File: .github/workflows/dockerimage.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+name: Docker Image
+on: 
+  push:
+    tags:
+      - '**'
+    branches:
+      - master
+  pull_request:
```
Comment (htdvisser) [inline .github/workflows/dockerimage.yml:8]: Not sure if it makes sense to do a 1h build for each pull request

**Review by htdvisser — APPROVED** [review]
It might be nice to do more work in parallel, but I think this is a good start.

File: .github/workflows/dockerimage.yml (hunk 1 above)
Comment (rvolosatovs) [reply .github/workflows/dockerimage.yml:8]: Only pull requests changing something image related.
Later we can enable caching for PR builds while performing clean `master` and tag builds.
This PR does not do anything new here compared to what already was done on Travis.

**Pull request merged**